packages/ui/src/server/controllers/clusters.ts

```diff
@@ -11,3 +11,35 @@ export function clusterVersions(req: Request, res: Response) {
         .then((data: {}) => sendResponse(res, data))
         .catch((error: any) => sendError(res, error));
 }
+
+type ClusterAuthStatus = Record<string, {authorized: boolean}>;
+
+export function clusterAuthStatus(req: Request, res: Response) {
+    const allowPasswordAuth = req.ctx.config.allowPasswordAuth;
+
+    let data = {};
+
+    if (allowPasswordAuth) {
+        data = Object.keys(req.cookies).reduce((ret, key) => {
+            if (key.includes(YT_CYPRESS_COOKIE_NAME)) {
+                const [cluster] = key.split(':');
```

Comment:
It is better to read cluster names from clusters-config.json, and check existance of only corresponding names.